docs/spec.md

```diff
@@ -1922,6 +1922,54 @@ imaginary values, `lhs` and `rhs`, and produces a `result` tensor.
 
 &nbsp;[More Examples](../stablehlo/tests/interpret/complex.mlir)
 
+### composite
+
+#### Semantics
+
+Encapsulates an operation made up (composed) of other StableHLO operations,
+taking `inputs` and producing `results`. The decomposition is specified
+using the `decomposition` attribute. The composite op can be replaced with
+its decomposition without changing semantics.
+
+The `version` and `decomposition_attributes` enable additional information
+to be communicated between producers and consumers. `version` defaults to 0
+and `decomposition_attributes` defaults to an empty dictionary.
+
+#### Inputs
+
+| Label | Name                                   | Type                                          |
+|-------|----------------------------------------|-----------------------------------------------|
+| (I1)  | `inputs`                               | variadic number of values                     |
+| (I2)  | `name`                                 | constant of type `string`                     |
+| (I3)  | `decomposition`                        | constant of type `function`                   |
+| (I4)  | `version`                              | constant of type `si32`                       |
+| (I5)  | `decomposition_attributes`             | attribute dictionary                          |
+
+#### Outputs
+
+| Name      | Type                      |
+|-----------|---------------------------|
+| `results` | variadic number of values |
+
+#### Constraints
+
+* (C1) `is_defined_in_parent_module(decomposition)`.
```

Comment:
The spec has not defined the concept of module. It is more related to how the functions in a program are organised.  Can we instead say that `decomposition` is defined in the same scope as the parent function of the op associated with decomposition.

Reply:
Thanks, good point! Done.